hail/src/main/scala/is/hail/expr/ir/MatrixValue.scala

```diff
@@ -338,5 +364,13 @@ object MatrixValue {
         BroadcastRow(RegionValue(ctx.r, rvb.end()), globalsPType, HailContext.backend),
         rvd))
   }
-
 }
+
+case class MatrixGlobalsEncoder(
```

Comment:
what's the null spec here?

I'm also a little unclear on how this fits into the broader plan. Could you write a paragraph on the rough steps to get to parallelized MatrixMultiWrite?

Reply:
We seem to treat globals as a table. So we have to write the their 'globals', which has no data. Since this is parametrized by a buffer spec, we need to compile an encoder for the empty physical struct.

I'll write more.